examples/systemcalls/systemcalls.c

```diff
@@ -58,10 +111,60 @@ bool do_exec(int count, ...)
  *   as second argument to the execv() command.
  *
 */
+    int status;
+    pid_t pid;
+    pid = fork ();
+    //handle if fork failed
+    if (pid == -1)
+    {
+        perror("Fork");
+        va_end(args);
+        return false;
+    }
+    //if child process
+    else if (pid == 0) 
+    {
+        if(execv (command[0], (command)) == -1)
+        {
+            //if exec returns, that means it failed
+            perror("Exec");
+            va_end(args);
+            exit(-1);    
+        }
```

Comment:
@GuruprashanthKrishnakumar I think it's safe to do this since execv shouldn't ever return anything other than -1 but for readability you could just always return error status when execv returns.  Otherwise it's ambiguous what would happen if it did for some reason return 0.